### PR TITLE
Extract manifest generation from WooCommerce docs plugin into a CLI tool

### PR DESCRIPTION
**Pull request by samueljseay**
The WooCommerce Docs plugin is designed to consume a basic manifest format. Up till now that prototype format was generated in a script within the plugin, but its desired to separate the tool out so it can be published and consumed from anywhere to generate manifest files from Markdown docs.

This extracts the logic for that script from the plugin into monorepo utils CLI and adds some supporting tests for the logic. This also updates instructions in README files so that you can generate a manifest file yourself in development in the WooCommerce docs plugin.

### Testing Instructions

1. Ensure that pnpm install has been run on a checkout of this branch
2. Run `pnpm build` on the monorepo utils package
3. From the root you can generate a manifest into the WooCommerce docs plugin as instructed [in the README](https://github.com/woocommerce/woocommerce/pull/39161/files#diff-e6483b5a793df3870167230500ed1be3fadf62c3cf8d59d411fba3d134bc5bbe) that comes with this PR and see that the manifest is generated correctly. 
4. Also see that tests run successfully via `pnpm test`, because we're still not running this packages tests along with the rest of the repo right now (something I intend to fix soon)

### CODE REVIEW

**Comment by samueljseay** [comment]
@psealock I hope you don't mind I pulled you in to review this one too. Its a PR into the other branch you're reviewing, so my plan would be after your approval to merge this to that PR then merge that PR as one into trunk.

🙇🏻

File: tools/monorepo-utils/src/md-docs/README.md
Comment (psealock) [inline tools/monorepo-utils/src/md-docs/README.md]: This sounds like it needs to be run from `/tools/monorepo-utils`

```suggestion
This command is built on postinstall and can be run from monorepo root.
```

File: tools/monorepo-utils/src/md-docs/commands/manifest/create.ts
Comment (psealock) [inline tools/monorepo-utils/src/md-docs/commands/manifest/create.ts]: It may be of no importance, but I'm wondering if this assumption that the `utils` command will always be run from monorepo root will hold up over time. We may package the utils up as an executable at some point. A few projects make use of `MONOREPO_ROOT` constant. We could consolidate these.

https://github.com/woocommerce/woocommerce/blob/0ae1fa1f5a5c82777b41e5e99f243583e07bc0c5/tools/monorepo-merge/src/const.ts#L7

File: tools/monorepo-utils/src/md-docs/commands/manifest/index.ts
Comment (psealock) [inline tools/monorepo-utils/src/md-docs/commands/manifest/index.ts]: This index file is in the `/manifest` folder. To stay consistent with the other tools, it would be in `/commands` and `create.ts` would be in `/commands/create`.

File: tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts
Comment (psealock) [inline tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts]: I needed to read the tests to figure out what each variable did. I think a jsDoc would be helpful here.

Aside: At first I would think `absoluteFilePath` below to be the exact same as the `filePath` argument, but if I'm thinking about this correctly, a relative path would also be accepted? If thats correct, another test would be appropriate.

**Review by psealock — COMMENTED** [review]
This working well and testing nicely. Good job @samueljseay, just a few comments and thoughts.

File: tools/monorepo-utils/src/md-docs/commands/manifest/create.ts
Comment (samueljseay) [reply tools/monorepo-utils/src/md-docs/commands/manifest/create.ts]: @psealock Thank you this certainly is important because I want to package up this CLI so other repos can use it to make manifests so yeah I definitely don't want to make that assumption. I've created an issue here: https://github.com/woocommerce/woocommerce/issues/39192

File: tools/monorepo-utils/src/md-docs/commands/manifest/index.ts
Comment (samueljseay) [reply tools/monorepo-utils/src/md-docs/commands/manifest/index.ts]: It seems I've broken away from the convention here with the Slack command and with this one. I can't remember my reasoning for doing that since they're functionally equivalent. I'll fix this one here and look to refactor the Slack one later.

Comment (samueljseay) [reply tools/monorepo-utils/src/md-docs/commands/manifest/index.ts]: Resolved in https://github.com/woocommerce/woocommerce/pull/39161/commits/f2adc16b045ac88f22d3cf7245eda55b7c2b4474

File: tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts
Comment (samueljseay) [reply tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts]: Yes you're right this whole thing was a mess and the params unclearly named. 

This function actually expects all paths to be absolute for it to work. it determines from a combination of root path and sub directory and absolute file path how to generate a url.

I've added some error throwing if you pass a relative path and updated the tests to cover those cases too. I also realized the order of params / naming of params in other functions was a mess too so I've reordered them or renamed them for clarity where it made sense: https://github.com/woocommerce/woocommerce/pull/39161/commits/3ed77ef2d967dc61863d79c00bce1c8cadb102d3

Also I noticed in the process of doing this that I'd accidentally left off generating of the hash the manifest should have so I added that back and have added a brief test for that too!

**Comment by samueljseay** [comment]
@psealock I think I addressed everything, keen for you to take another look!

**Review by psealock — APPROVED** [review]
Thanks for the changes @samueljseay, looking good and working well 🚀

**Pull request merged**
